numpy/ma/core.pyi

```diff
@@ -1169,7 +1169,17 @@ class MaskedArray(ndarray[_ShapeT_co, _DTypeT_co]):
 
     def nonzero(self) -> tuple[_Array1D[intp], ...]: ...
     def trace(self, offset=..., axis1=..., axis2=..., dtype=..., out=...): ...
-    def dot(self, b, out=..., strict=...): ...
+
+    # Keep in sync with `ndarray.dot`.
+    # 1D + 1D returns a scalar;
+    # all other with at least 1 non-0D array return MaskedArray.
+    @overload
+    def dot(self, b: _ScalarLike_co, out: None = ..., strict: bool = ...) -> _MaskedArray[Any]: ...
+    @overload
+    def dot(self, b: ArrayLike, out: None = ..., strict: bool = ...) -> Any: ...  # type: ignore[misc]
```

Comment:
apparently this `dot` differs from the one of `ndarray`:

```pycon
>>> import numpy as np
>>> np.array([1,2]).dot(np.array([3, 2]))
np.int64(7)
>>> np.ma.array([1,2]).dot(np.ma.array([3, 2]))
masked_array(data=7,
             mask=False,
       fill_value=999999)
>>> np.ma.array([1,2]).dot(np.array([3, 2]))
masked_array(data=7,
             mask=False,
       fill_value=999999)
>>> np.array([1,2]).dot(np.ma.array([3, 2]))
np.int64(7)
```

so `1D dot 1D` doesn't return a scalar here, which makes it easier to type :)